src/Native/System.Security.Cryptography.Native/pal_pkcs7.cpp

```diff
@@ -52,3 +71,23 @@ extern "C" int32_t GetPkcs7Certificates(PKCS7* p7, X509Stack** certs)
 
     return 0;
 }
+
+extern "C" int32_t Pkcs7AddCertificate(PKCS7* p7, X509* x509)
+{
+    if (p7 == nullptr)
```

Comment:
Should x509 be checked here as well?  I know there's a pattern around allowing values returned from one API to be passed directly to the next... not sure how many arguments that applies to.